### PR TITLE
Ensure has_field() supports dynamic fields

### LINKED ISSUE #3346: [BUG] Document `has_field` does not work for non-default field names
### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 16.04): Ubuntu 20.04.3 LTS
-   **Python version** (`python --version`): Python 3.10.2
-   **FiftyOne version** (`fiftyone --version`): 0.21.4
-   **FiftyOne installed from** (pip or source): pip

### Commands to reproduce

Running the following in an `ipython` shell should demonstrate the problem with `has_field`. 

```
import fiftyone as fo
# Create an empty detection
det = fo.Detection()
# Add a custom field to the detection
det['asdf'] = 1.0
# Check with has_field
print(f"Has asdf: {det.has_field('asdf')}")
# Check with `in`
print(f"Has asdf (in): {'asdf' in det}")
```

### Describe the problem

In previous versions, custom fields on a `Detection` would work with `has_field`.  Since updating to version 0.21.4, this no longer works for detections regardless of saved status.

### Code to reproduce issue

See above.  Can be demonstrated with the quickstart dataset as well after saving samples

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core Python library issue
-   [ ] `Server`: FiftyOne server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [x ] Yes. I would be willing to contribute a fix for this bug with guidance from the FiftyOne community
-   [ ] No. I cannot contribute a bug fix at this time

### PR DESCRIPTION
**Pull request by brimoor**
Resolves #3346

### CODE REVIEW

File: fiftyone/core/odm/document.py [hunk 1]
```diff
@@ -297,16 +297,15 @@ def field_names(self):
...
-        if len(chunks) > 1:
-            field = self._fields.get(chunks[0], None)
-            try:
-                return field.has_field(chunks[1])
-            except AttributeError:
-                return False
+        try:
+            chunks = field_name.split(".", 1)
+            if len(chunks) > 1:
+                return self._get_field(chunks[0]).has_field(chunks[1])
```
Comment (swheaton) [inline fiftyone/core/odm/document.py:303]:
```suggestion
```
I think you can rely on your new `self._get_field(field_name)` to do this initial split for you.

**Review by swheaton — CHANGES_REQUESTED** [review]
lgtm and seems to work as advertised!
Just one minor comment

File: fiftyone/core/odm/document.py (hunk 1 above)
Comment (brimoor) [reply fiftyone/core/odm/document.py:303]: Good call, will add 👍

**Review by swheaton — APPROVED** [review]
lgtm

**Pull request merged**